### PR TITLE
Bump test dependencies, Scala 2.12.0-M3

### PR DESCRIPTION
**Pull request by seratch**

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -41,13 +41,13 @@ object ScalikeJDBCProjects extends Build {
...
-        case _ => "2.2.5-M2"
+        case Some((2, v)) if v <= 11 => "2.2.6"
+        case _ => "2.2.5-M3"
       }
     },
     specs2Version := {
       CrossVersion.partialVersion(scalaVersion.value) match {
-        case Some((2, v)) if v <= 11 => "2.4.17"
+        case Some((2, v)) if v <= 11 => "2.5"
         case _ => "3.6.3"
```
Comment (xuwei-k) [inline project/Build.scala]: http://repo1.maven.org/maven2/org/specs2/specs2-core_2.12.0-M3/

`"org.specs2" % "specs2-core_2.12.0-M3" % "3.6.3"` does not exists

**Comment by xuwei-k** [comment]
https://travis-ci.org/scalikejdbc/scalikejdbc/jobs/100497511#L701

```
[error] Modules were resolved with conflicting cross-version suffixes in {file:/home/travis/build/scalikejdbc/scalikejdbc/}test:
[error]    org.scalaz:scalaz-effect _2.12.0-M3, _2.12.0-M2
[error]    org.scala-lang.modules:scala-xml _2.12.0-M3, _2.12.0-M2
[error]    org.scalaz:scalaz-concurrent _2.12.0-M3, _2.12.0-M2
[error]    org.scalaz:scalaz-core _2.12.0-M3, _2.12.0-M2
[error]    org.scala-lang.modules:scala-parser-combinators _2.12.0-M3, _2.12.0-M2
[error] Modules were resolved with conflicting cross-version suffixes in {file:/home/travis/build/scalikejdbc/scalikejdbc/}mapper-generator-core:
[error]    org.scalaz:scalaz-effect _2.12.0-M3, _2.12.0-M2
[error]    org.scala-lang.modules:scala-xml _2.12.0-M3, _2.12.0-M2
[error]    org.scalaz:scalaz-concurrent _2.12.0-M3, _2.12.0-M2
[error]    org.scalaz:scalaz-core _2.12.0-M3, _2.12.0-M2
[error]    org.scala-lang.modules:scala-parser-combinators _2.12.0-M3, _2.12.0-M2
java.lang.RuntimeException: Conflicting cross-version suffixes in: org.scalaz:scalaz-effect, org.scala-lang.modules:scala-xml, org.scalaz:scalaz-concurrent, org.scalaz:scalaz-core, org.scala-lang.modules:scala-parser-combinators
	at scala.sys.package$.error(package.scala:27)
	at sbt.ConflictWarning$.processCrossVersioned(ConflictWarning.scala:46)
	at sbt.ConflictWarning$.apply(ConflictWarning.scala:32)
	at sbt.Classpaths$$anonfun$69.apply(Defaults.scala:1179)
	at sbt.Classpaths$$anonfun$69.apply(Defaults.scala:1176)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
```

**Comment by seratch** [comment]
I'm not sure how to fix this yet. Although testing with 2.12.0-M3 is not so important for me at this time.

**Comment by xuwei-k** [comment]
`specs2-core_2.12.0-M3` depends on `scalaz-stream_2.12.0-M2` due to https://github.com/functional-streams-for-scala/fs2/pull/463

**Comment by seratch** [comment]
Thanks. Your service is quite useful.

**Comment by xuwei-k** [comment]
https://github.com/etorreborre/specs2/issues/425#issuecomment-182554642

**Comment by seratch** [comment]
Updated my changes and cleared Travis cache (specs2-core overwrote the same version 3.7).

**Comment by seratch** [comment]
Builds succeeded.

**Pull request merged**